EdurecaInClassProject/src/app/components/select-seat/select-seat.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit } from '@angular/core';
+import { BusService } from '../../services/bus.service';
+import { Seat } from '../../models/seat.model';
+import {Router, ActivatedRoute} from '@angular/router';
+
+@Component({
+  selector: 'app-select-seat',
+  templateUrl: './select-seat.component.html',
+  styleUrls: ['./select-seat.component.css']
+})
+export class SelectSeatComponent implements OnInit {
+
+  showSeatList:Seat[]=[];
+  total:number=0;
+  busId:number=0;
+  fare:number=0;
+  alert:boolean=false;
+  fillupSeat:any[]=[];
+
+  constructor(private _busService:BusService,private router : Router, private route : ActivatedRoute) {
+   }
+
+  ngOnInit(): void {
+    this.route.params.subscribe((params) => {
+      this.busId=parseInt(params.id);
+    })
+  }
+
+  seat(s){
+     let seats;
+     this.fare=this._busService.getFare(this.busId);
+     seats= this.showSeatList.map(iteam=>{
+     return iteam.seatNo
```

Comment:
we don't need return statement here

Reply:
okay I will do the changes

Reply:
that return statement is needed there i am using a mapping function.